frontend/src/util/types/Requisition.ts

```diff
@@ -0,0 +1,139 @@
+import {User} from "./User";
+import {Project} from "./Project";
+import {Vendor} from "./Vendor";
+import {gql} from "@apollo/client";
+import {SemanticCOLORS} from "semantic-ui-react/dist/commonjs/generic";
+
+export type RequisitionStatus =
+    "DRAFT" |
+    "SUBMITTED" |
+    "PENDING_CHANGES" |
+    "READY_TO_ORDER" |
+    "ORDERED" |
+    "RECEIVED" |
+    "CANCELLED";
+
+export const StatusToColor = (status: RequisitionStatus): SemanticCOLORS => {
+    switch (status) {
+        case "DRAFT": return "grey"
+        case "SUBMITTED": return "yellow"
+        case "PENDING_CHANGES": return "orange"
+        case "READY_TO_ORDER": return "green"
+        case "ORDERED": return "blue"
+        case "RECEIVED": return "blue"
+        case "CANCELLED": return "red"
+    }
+}
+
+export const StatusToString = (status: RequisitionStatus): string => {
+    switch (status) {
+        case "DRAFT": return "Draft"
+        case "SUBMITTED": return "Submitted"
+        case "PENDING_CHANGES": return "Pending Changes"
+        case "READY_TO_ORDER": return "Ready to Order"
+        case "ORDERED": return "Ordered"
+        case "RECEIVED": return "Received"
+        case "CANCELLED": return "Cancelled"
+    }
+}
+
+export type Requisition = {
+    id: number,
+    headline: string,
+    description: string,
+    status: RequisitionStatus,
+    pointOfContact: User,
+    createdBy: User,
+    project: Project,
+    vendor: Vendor,
+    projectRequisitionId: number,
+    paymentRequiredBy: Date,
+    requisitionitemSet: RequisitionItem[],
+    referenceString: string,
+    canEdit: boolean,
+    otherFees: number
+}
+
+// Sums up total costs for requisition items
+export const getTotalItemsCost = (requisition: Requisition): string => {
+    let total = 0;
+    for (const requisitionItem of requisition.requisitionitemSet) {
+        total += requisitionItem.quantity * requisitionItem.unitPrice
+    }
+
+    return total.toFixed(2);
+}
+
+// Sums up total costs for requisition items plus other fees
+export const getTotalCost = (requisition: Requisition): string => {
+    let total = 0;
+    for (const requisitionItem of requisition.requisitionitemSet) {
+        total += requisitionItem.quantity * requisitionItem.unitPrice
+    }
+
+    total += requisition.otherFees;
+
+    return total.toFixed(2)
+}
+
+export type RequisitionItem = {
+    name: string,
+    unitPrice: number,
+    quantity: number,
+    link: string,
+    notes: string
+}
+
+export const OPEN_REQUISITIONS_QUERY = gql`
+    query requisitions {
```

Comment:
With Apollo, you need to be careful to be querying a common set of elements about GraphQL types - read more about caching starting with this page https://www.apollographql.com/docs/react/v3.0-beta/caching/cache-configuration/

This generally comes into play when you're doing any kind of mutation because Apollo maintains a local cache that has to be updated when the mutation is executed.  If you install the Apollo dev tools extension in Chrome, it will show you a visual representation of the cache, which can be helpful for understanding what's happening under the hood.

[GraphQL fragments](https://www.apollographql.com/docs/react/v3.0-beta/data/fragments/) might be worth looking into to help with this

Reply:
Will be fixed in next pull when adding mutations.